### PR TITLE
feature: response-rewrite plugin rewrite binary data to client by base64

### PR DESCRIPTION
**Pull request by lilien1010**
### Summary

response-rewrite plugin support rewrite binary data to client

### Full changelog

* doc/plugins/response-rewrite-cn.md
* doc/plugins/response-rewrite.md
* apisix/plugins/response-rewrite.lua
* apisix/t/plugin/response-rewrite.t

### CODE REVIEW

File: apisix/plugins/response-rewrite.lua [hunk 1]
```diff
@@ -85,7 +90,13 @@ do
 
 function _M.body_filter(conf, ctx)
     if conf.body then
-        ngx.arg[1] = conf.body
+
+        if conf.body_base64 then
+            ngx.arg[1] = ngx.decode_base64(conf.body)
```
Comment (moonming) [inline apisix/plugins/response-rewrite.lua:95]: if conf.body is not well formed, will return `nil`. we need add test case for it.

File: apisix/plugins/response-rewrite.lua [hunk 2]
```diff
@@ -33,6 +33,11 @@ local schema = {
             description = "new body for repsonse",
             type = "string",
         },
+        body_base64 = {
```
Comment (membphis) [inline apisix/plugins/response-rewrite.lua:36]: we can check if it is a valid `base64` value in `function _M.check_schema(conf)`.

Comment (lilien1010) [reply apisix/plugins/response-rewrite.lua:36]: done

File: apisix/plugins/response-rewrite.lua (hunk 1 above)
Comment (lilien1010) [reply apisix/plugins/response-rewrite.lua:102]: done

**Pull request merged**